app/controllers/HomePageController.scala

```diff
@@ -46,23 +46,29 @@ class HomePageController @Inject() (
       doesGoodsRecordExist = goodsRecords.exists(_.goodsItemRecords.nonEmpty)
     } yield {
       val downloadLinkMessagesKey = getDownloadLinkMessagesKey(downloadDataSummary, doesGoodsRecordExist)
-      downloadDataSummary match {
-        case Some(downloadDataSummary) if downloadDataSummary.status == FileReadyUnseen =>
-          Ok(view(downloadReady = true, downloadLinkMessagesKey))
-        case _                                                                          =>
-          Ok(view(downloadReady = false, downloadLinkMessagesKey))
-      }
+      Ok(view(downloadReady(downloadDataSummary), downloadLinkMessagesKey))
     }
   }
 
-  private def getDownloadLinkMessagesKey(opt: Option[DownloadDataSummary], doesGoodsRecordExist: Boolean): String =
+  private def downloadReady(downloadDataSummary: Option[Seq[DownloadDataSummary]]): Boolean =
+    downloadDataSummary
+      .flatMap(
+        _.collectFirst {
```

Comment:
why are we collecting the first? don't we just want to see if any of the download data summaries have FileReadyUnseen and then show the banner if that is the case?

Reply:
collectFirst just stops on the first occurring instance of FileReadyUnseen, so if any are contained in the list the banner will show, if none in the list None will be returned which is then turned into false via the .getOrElse(false)